src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/barcode/BarcodeEventRepository.kt

```diff
@@ -17,8 +18,9 @@ import javax.validation.constraints.NotNull
 
 @Repository
 interface BarcodeEventRepository : JpaRepository<BarcodeEvent, Long> {
-  fun findByBarcode(barcode: Barcode): List<BarcodeEvent>
   fun findByBarcodeAndStatusOrderByCreatedDateTime(barcode: Barcode, status: BarcodeStatus): List<BarcodeEvent>
+  @Query("select be from BarcodeEvent as be where be.barcode = :barcode and be.status = 'CREATED'")
+  fun findByBarcodeAndStatusCreated(barcode: Barcode): BarcodeEvent?
```

Comment:
We only ever expect 0 or 1 created events - if this isn't true better to blow up in the query (it should never happen)

Reply:
Minor, and I dont know if this works with spring respository interfaces, but could the return type be `Optional<BarcodeEvent>` ? Then it would be really clear that we expect 1 or 0
Just a thought (and like I said, I dont even know if that works 😁 )

Reply:
The whole point of nullable in Kotlin is that it's better than Optional! The question mark indicatest it's nullable and I'm glad to see the back of Optional tbh.